src/Mvc/Mvc.Core/src/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -183,11 +183,24 @@ public async Task BindModelAsync(ModelBindingContext bindingContext)
                 // If instead the input formatter wants to treat the input as optional, it must do so by
                 // returning InputFormatterResult.Success(defaultForModelType), because input formatters
                 // are responsible for choosing a default value for the model type.
-                var message = bindingContext
-                    .ModelMetadata
-                    .ModelBindingMessageProvider
-                    .MissingRequestBodyRequiredValueAccessor();
-                bindingContext.ModelState.AddModelError(modelBindingKey, message);
+
+                // If request content length is 0 then request had empty body, if not null was provided in json
+                if (httpContext.Request.ContentLength == 0)
```

Comment:
I don't feel that this is the right place to do this. Maybe an option is change the json-based inpur formatters to AddModelError and return Failure instead

https://github.com/dotnet/aspnetcore/blob/1c5da276ce957e9db063d7dc7440ea6cc496300c/src/Mvc/Mvc.NewtonsoftJson/src/NewtonsoftJsonInputFormatter.cs#L206

https://github.com/dotnet/aspnetcore/blob/1c5da276ce957e9db063d7dc7440ea6cc496300c/src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonInputFormatter.cs#L111

Please check if we do have any side effects with this change.

Reply:
Duplicated comment.